rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/obligation/ObligationController.java

```diff
@@ -0,0 +1,118 @@
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.linkTo;
+
+@BasePathAwareController
+@Slf4j
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class ObligationController implements ResourceProcessor<RepositoryLinksResource> {
+    public static final String OBLIGATION_URL = "/obligations";
+
+    @NonNull
+    private final Sw360ObligationService obligationService;
+
+    @NonNull
+    private final RestControllerHelper restControllerHelper;
+
+    @RequestMapping(value = OBLIGATION_URL, method = RequestMethod.GET)
+    public ResponseEntity<Resources<Resource<Obligation>>> getObligations() {
+        List<Obligation> obligations = obligationService.getObligations();
+
+        List<Resource<Obligation>> obligationResources = new ArrayList<>();
+        obligations.forEach(o -> {
+            Obligation embeddedObligation = restControllerHelper.convertToEmbeddedObligation(o);
+            obligationResources.add(new Resource<>(embeddedObligation));
+        });
+
+        Resources<Resource<Obligation>> resources = new Resources<>(obligationResources);
+        return new ResponseEntity<>(resources, HttpStatus.OK);
+    }
+
+    @RequestMapping(value = OBLIGATION_URL + "/{id}", method = RequestMethod.GET)
+    public ResponseEntity<Resource<Obligation>> getObligation(
+            @PathVariable("id") String id) {
+        Obligation sw360Obligation = obligationService.getObligationById(id);
+        HalResource<Obligation> halResource = createHalObligation(sw360Obligation);
+        return new ResponseEntity<>(halResource, HttpStatus.OK);
+    }
+
+    @PreAuthorize("hasAuthority('WRITE')")
+    @RequestMapping(value = OBLIGATION_URL, method = RequestMethod.POST)
+    public ResponseEntity createObligation(
+            @RequestBody Obligation obligation) {
+        User sw360User = restControllerHelper.getSw360UserFromAuthentication();
+        obligation = obligationService.createObligation(obligation, sw360User);
+        HalResource<Obligation> halResource = createHalObligation(obligation);
+
+        URI location = ServletUriComponentsBuilder
+                .fromCurrentRequest().path("/{id}")
+                .buildAndExpand(obligation.getId()).toUri();
+
+        return ResponseEntity.created(location).body(halResource);
+    }
+
+    @PreAuthorize("hasAuthority('WRITE')")
+    @RequestMapping(value = OBLIGATION_URL + "/{ids}", method = RequestMethod.DELETE)
+    public ResponseEntity<List<MultiStatus>> deleteObligations(
+            @PathVariable("ids") List<String> idsToDelete) throws TException {
+        User user = restControllerHelper.getSw360UserFromAuthentication();
+        List<MultiStatus> results = new ArrayList<>();
+        for(String id:idsToDelete) {
```

Comment:
please correct the indentation

Reply:
Sorry, do you mean the indentation of line 95?
`            @PathVariable("ids") List<String> idsToDelete) throws TException {`
Checking other files, I think this is the coding convention of SW360.

Reply:
It was in the for loop, which looks good now